static/js/message_events.js

```diff
@@ -187,19 +188,24 @@ exports.update_messages = function update_messages(events) {
                     if (current_filter && current_filter.has_topic(stream_name, orig_topic)) {
                         let new_filter = current_filter;
                         if (new_filter && stream_changed) {
-                            // TODO: This logic doesn't handle the
-                            // case where we're a guest user and the
-                            // message moves to a stream we cannot
-                            // access, which would cause the
-                            // stream_data lookup here to fail.
-                            //
-                            // The fix is likely somewhat involved, so punting for now.
-                            const new_stream_name = stream_data.get_sub_by_id(new_stream_id).name;
-                            new_filter = new_filter.filter_with_new_params({
-                                operator: 'stream',
-                                operand: new_stream_name,
-                            });
-                            changed_narrow = true;
+                            if (new_stream === undefined) {
+                                // The new stream, where the messages were moved, isn't known
+                                // to the user possibly because it is a guest user.
+                                // Here we just narrow to the old stream, or unnarrow
+                                // from the old stream > topic.
+                                new_filter = new Filter([{
+                                    operator: 'stream',
+                                    operand: stream_name,
+                                }]);
+                                changed_narrow = true;
```

Comment:
Do we want to renarrow in this case, or just leave the view as-is, in this unusual case?  I guess moving to the stream narrow is inoffensive.  But why does it mention "unnarrow" in the comment?